### DISCUSSION TITLE
Great progress, but much to do!

### DISCUSSION THREAD

**Discussion by coverbeek** [frc5567/2023-Robot#219] [Team Posts]
Thanks to everyone for their contributions today and thus far. Our codebase is looking good and we have a good foundation to build on. 

Thanks to @HannahStrimpel for pushing hard on the vision front and getting some of our younger coders spinning up on PID.

Thanks to @CalebOverbeek for pushing hard on developing an Auton and also getting some of the younger coders thinking on that.

Thanks to @sam-halliday and @programmeruser517 for pushing forward on finalizing the last of the core classes needed for control.

I don’t want to diminish what has been done, but now the hard stuff has to be done. We need to transition from copy and paste of past code to making everything work. This is when we make it all come together. This is also when we need everyone not only learning, but truly understanding the code we’re writing. 
I’ve been coding since I was 12 and thats a really long time ago - I’m still learning. I want every one of you to have a mindset of learning - every day. That’s what it is to be a coder/programmer/engineer.
I encourage the seniors to share links to what they’re studying and learning from. I’ll share my current favorite: https://docs.wpilib.org/en/stable/docs/software/pathplanning/index.html

**Comment by HannahStrimpel** [comment]
Here is the video we were using to review PID for everyone else to review: [https://www.youtube.com/watch?v=-KNaluv9dd8](https://www.youtube.com/watch?v=-KNaluv9dd8)

**Comment by HannahStrimpel** [comment]
https://www.youtube.com/watch?v=-KNaluv9dd8

**Comment by HannahStrimpel** [comment]
^^ Use 2nd link. The 1st one doesn't work.
